src/wp-includes/comment-template.php

```diff
@@ -1942,7 +1942,14 @@ function get_comment_id_fields( $post_id = 0 ) {
 
 	$reply_to_id = isset( $_GET['replytocom'] ) ? (int) $_GET['replytocom'] : 0;
 	$result      = "<input type='hidden' name='comment_post_ID' value='$post_id' id='comment_post_ID' />\n";
-	$result     .= "<input type='hidden' name='comment_parent' id='comment_parent' value='$reply_to_id' />\n";
+
+	if ( 0 !== $reply_to_id ) {
+		$comment = get_comment( $reply_to_id );
+
+		if ( 0 !== (int) $comment->comment_approved && $post_id === (int) $comment->comment_post_ID ) {
+			$result .= "<input type='hidden' name='comment_parent' id='comment_parent' value='$reply_to_id' />\n";
+		}
+	}
```

Comment:
The `#comment_parent` element always needs to be rendered as the default [comment-reply JavaScript](https://github.com/WordPress/wordpress-develop/blob/7164e4b4c8f1a64cd9e558ca9fa41d35c714eb6b/src/js/_enqueues/lib/comment-reply.js) expects it.

```suggestion
	if ( 0 !== $reply_to_id ) {
		$comment = get_comment( $reply_to_id );

		if ( 0 !== (int) $comment->comment_approved && $post_id === (int) $comment->comment_post_ID ) {
			$reply_to_id = 0;
		}
	}

	$result .= "<input type='hidden' name='comment_parent' id='comment_parent' value='$reply_to_id' />\n";
```

Reply:
I'm going to update the PR based on this and include a null check on `$comment`, etc. as well.